src/libsync/propagateuploadng.cpp

```diff
@@ -0,0 +1,401 @@
+                                                    &_propagator->_anotherSyncNeeded);
+        QString errorString = errorMessage(job->reply()->errorString(), job->reply()->readAll());
+        if (job->reply()->hasRawHeader("OC-ErrorString")) {
+            errorString = job->reply()->rawHeader("OC-ErrorString");
+        }
+        abortWithError(status, errorString);
+        return;
+    }
+    startNextChunk();
+}
+
+void PropagateUploadFileNG::startNextChunk()
+{
+    if (_propagator->_abortRequested.fetchAndAddRelaxed(0))
+        return;
+
+    quint64 fileSize = _item->_size;
+
+    quint64 currentChunkSize = qMin(chunkSize(), fileSize - _sent);
+
+    if (currentChunkSize <= 0) {
+        Q_ASSERT(_jobs.isEmpty()); // There should be no running job anymore
+        _finished = true;
+        // Finish with a MOVE
+        QString destination = _propagator->_remoteDir + _item->_file;
+        auto headers = PropagateUploadFileCommon::headers();
+        auto job = new MoveJob(_propagator->account(), Account::concatUrlPath(chunkUrl(), "/.file"),
+                               destination, headers, this);
+        _jobs.append(job);
+        connect(job, SIGNAL(finishedSignal()), this, SLOT(slotMoveJobFinished()));
+        connect(job, SIGNAL(destroyed(QObject*)), this, SLOT(slotJobDestroyed(QObject*)));
+        _propagator->_activeJobList.append(this);
+        job->start();
+        return;
+    }
+
+    auto device = new UploadDevice(&_propagator->_bandwidthManager);
+    const QString fileName = _propagator->getFilePath(_item->_file);
+
+    if (! device->prepareAndOpen(fileName, _sent, currentChunkSize)) {
+        qDebug() << "ERR: Could not prepare upload device: " << device->errorString();
+
+        // If the file is currently locked, we want to retry the sync
+        // when it becomes available again.
+        if (FileSystem::isFileLocked(fileName)) {
+            emit _propagator->seenLockedFile(fileName);
+        }
+        // Soft error because this is likely caused by the user modifying his files while syncing
+        abortWithError( SyncFileItem::SoftError, device->errorString() );
+        return;
+    }
+
+    _sent += currentChunkSize;
+    QUrl url = chunkUrl(_currentChunk);
+
+    // job takes ownership of device via a QScopedPointer. Job deletes itself when finishing
+    PUTFileJob* job = new PUTFileJob(_propagator->account(), url, device, {}, _currentChunk);
+    _jobs.append(job);
+    connect(job, SIGNAL(finishedSignal()), this, SLOT(slotPutFinished()));
+    connect(job, SIGNAL(uploadProgress(qint64,qint64)),
+            this, SLOT(slotUploadProgress(qint64,qint64)));
+    connect(job, SIGNAL(uploadProgress(qint64,qint64)),
+            device, SLOT(slotJobUploadProgress(qint64,qint64)));
+    connect(job, SIGNAL(destroyed(QObject*)), this, SLOT(slotJobDestroyed(QObject*)));
+    job->start();
+    _propagator->_activeJobList.append(this);
+    _currentChunk++;
+
+    // FIXME! parallel chunk?
```

Comment:
It's sad to lose the feature, but it's also nice to have less complexity. Maybe if the scaling to a bigger chunk size is good enough you don't need to do it.